gax-java/gax/src/main/java/com/google/api/gax/rpc/StubSettings.java

```diff
@@ -137,6 +140,13 @@ public final ApiClock getClock() {
     return clock;
   }
 
+  // Intended for Internal Use and Overriden by generated ServiceStubSettings classes.
+  // Meant to be shared between StubSettings and ClientContext.
+  @InternalApi
+  public String getServiceName() {
+    return "";
+  }
```

Comment:
`ClientContext.create()` takes in the root StubSettings (not the subclasses). Create a default method that generated subclasses can override.